### PR TITLE
Calculates duration for extended bolus that was cancelled

### PR DESCRIPTION
**Pull request by gniezen**
- Used the time of the "Extended Bolus Completed" (`LID_BOLEX_COMPLETED`) event to determine the actual duration of the extended bolus in the case where it was cancelled. 
- Refactored the extended bolus code to make it a bit clearer.

### CODE REVIEW

**Comment by gniezen** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;Ar8ocrtj&#x2F;21-1-tandem-boluses-clean-up"> &lt;1&gt; [tandem] boluses clean-up</a>

**Comment by jebeck** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;hnLiq6I3&#x2F;22-1-tandem-wizards-clean-up"> &lt;1&gt; [tandem] wizards clean-up</a>

File: lib/drivers/tandemTslimDriver.js [hunk 1]
```diff
@@ -1262,50 +1309,73 @@ module.exports = function (config) {
...
           }
+          else {
+            record = record.with_extended(bolus.bolex_insulin_delivered);
+          }
           record = record.with_expectedExtended(bolus.bolex_size)
             .with_expectedDuration(bolus.duration);
         }
+        // other case: extended bolus completed
+        else {
+          record = record.with_extended(bolus.insulin_delivered);
```
Comment (jebeck) [inline lib/drivers/tandemTslimDriver.js]: Careful: this is wrong, should be `bolus.bolex_insulin_delivered`.

**Comment by jebeck** [comment]
Merging for now. May be additional clean-up when I have access to more data to audit, but that will have to wait for #175

**Pull request merged**